### ISSUE TITLE
How to upgrade Supermarket?

### ISSUE THREAD

**Issue by legal90** [chef/supermarket#1392]
I have a question about upgrading private Supermarket instance.
* I have Supermarket [`v1.7.0.alpha`](https://github.com/chef/supermarket/tree/1.7.0-alpha.0) deployed long time ago with ["supermarket" cookbook](https://github.com/chef-cookbooks/supermarket), which is deprecated now.
* I wanna upgrade it to the latest version, (v2.8.1 at the moment) with actual cookbook ["supermarket-omnibus-cookbook"](https://github.com/chef-cookbooks/supermarket-omnibus-cookbook)

How can I do it smoothly, without data loss?

**Comment by nellshamrell** [comment]
Hi @legal90!

Do you have your database, cache, and cookbook storage on the same server as your supermarket server?  Or are they in separate places (i.e. having a Supermarket server in EC2, database in RDS, cache in Elasticache, and Cookbook storage in S3)?

**Comment by legal90** [comment]
Hi, @nellshamrell 

Yes, I have a simple installation with database, cache, and cookbook storage located on the same server as supermarket app.

And seems like I found a proper way for upgrading Supermarket in my case. There is a short manual:

**1)** Deploy the current (old) Supermarket version on the new host using the actual cookbook `supermarket-omnibus-cookbook`. There is no "1.7.0-alpha.0" on the "stable" channel by some reason, so I've took "1.8.0-alpha.0":
```ruby
default['supermarket_omnibus']['package_version'] = '1.8.0-alpha.0'
```

**2)** Copy the cookbook storage from old Supermarket instance to the new one. Default paths are:
  * Current instance, deployed by _"supermarket"_ cookbook: `/srv/supermarket/shared/system/cookbook_versions`
  * New instance, deployed by _"supermarket-omnibus-cookbook"_ : `/var/opt/supermarket/data/cookbook_versions`

NB! This dir should be writable for "supermarket" user.

**3)** Createa dump of the current supermarket database and restore it on the new instance:
```console
[old_host] $ pg_dump <supermaket_database_name> -c -f supermarket.sql

[new_host] $ /opt/supermarket/embedded/bin/psql -h 127.0.0.1 -p 15432 supermarket -f /path/to/supermarket.sql
```
**4)** At this step, the new instance should work fine and be a full copy of your old-style deployed Supermarket. You will probably need to update DNS entries and/or double-check you OAuth (oc-id) configuration to make sure that authorization through your Chef Server works fine.

**5)** Upgrade you newly deployed Supermarket to the latest version by setting an attribute and running the Chef Client:
```ruby
default['supermarket_omnibus']['package_version'] = '2.8.1' # or :latest
```
**6)** Done!

I've reproduced all these steps in TestKitchen instance and everything works fine. Gonna do this on production and will let you now soon. 😃
